components/Input/react/Input.tsx

```diff
@@ -0,0 +1,125 @@
+import * as React from 'react'
+import clsx from 'clsx'
+import {
+  InputProps,
+  DefaultSize,
+  DefaultVariant,
+  SizeClassesTable,
+  VariantClassesTable,
+  IconStaticClasses,
+  ResetStaticClasses,
+  ResultStaticClass,
+  StaticClasses,
+  inputClasses,
+  StaticInputClasses,
+} from '@cypress-design/constants-input'
+import {
+  IconObjectMagnifyingGlass,
+  IconActionDeleteLarge,
+} from '@cypress-design/react-icon'
+
+export interface InputPropsJsx extends InputProps {
+  id: string
+  value?: string
+  className?: string
+}
+
+type ReactInputProps = InputPropsJsx & React.HTMLProps<HTMLInputElement>
+
+// tbd: what is a better way of allowing the <input/>
+//  to propagate its focus state to siblings?
+const usePropagateFocusProps = () => {
+  const [hasFocus, setHasFocus] = React.useState(false)
+  const enableFocusStyle = React.useCallback(() => setHasFocus(true), [])
+  const disableFocusStyle = React.useCallback(() => setHasFocus(false), [])
+  const propagateFocusProps = React.useMemo(() => {
+    return {
+      onFocus: enableFocusStyle,
+      onMouseEnter: enableFocusStyle,
+      onMouseOver: enableFocusStyle,
+      onBlur: disableFocusStyle,
+      onMouseLeave: disableFocusStyle,
+      onMouseOut: disableFocusStyle,
+    }
+  }, [enableFocusStyle, disableFocusStyle])
+  return {
+    propagateFocusProps,
+    hasFocus,
+  }
+}
```

Comment:
Changing the style on focus is usually easier to build using tools like `focus-within`.

Reply:
Is there an example of it in a react component at this time, it seems you've only done it in vue atm?

Reply:
it is actually pure css... I'll modify this with an example.